### PR TITLE
Refactoring the event setting processing

### PR DESCRIPTION
**Pull request by tsuyoshiwada**
Minor refactoring.

### CODE REVIEW

File: src/supports.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// @flow
+
+const win = window;
```
Comment (oliviertassinari) [inline src/supports.js]: I think it's going to break the server-side rendering.

Comment (tsuyoshiwada) [inline src/supports.js]: Confirmation had leaked...
I'll revise this mistakes.

**Comment by oliviertassinari** [comment]
I'm curious, what's the motivation behind a new `supports` module?

**Comment by tsuyoshiwada** [comment]
Thanks for fast reply.

If add the passive option.
That's because I thought that keep to simple components file by placing one of the logic if it can use the external.

**Comment by tsuyoshiwada** [comment]
Fixed an error in the server-side rendering. ([eb66649](https://github.com/oliviertassinari/react-event-listener/pull/12/commits/eb66649c2e39d23ff1a9a97374f84cb43d8fb346))

**Comment by oliviertassinari** [comment]
That looks good. Thanks 👍 .

**Comment by tsuyoshiwada** [comment]
Thanks!

**Pull request merged**